### ISSUE TITLE
ArchGDAL randomly infers wrong file type

### ISSUE THREAD

**Issue by adamchlus** [emit-sds/SpectralUnmixing#5]
I was unmixing a bunch of ISOFIT generated reflectance images and randomly on a single image got the following error:

```
ERROR: GDALError (CE_Failure, code 1):
	Driver ARCGEN is considered for removal in GDAL 3.5. You are invited to convert any dataset in that format to another more common one. If you need this driver in future GDAL versions, create a ticket at https://github.com/OSGeo/gdal (look first for an existing one first) to explain how critical it is for you (but the GDAL project may still remove it), and to enable it now, set the GDAL_ENABLE_DEPRECATED_DRIVER_ARCGEN configuration option / environment variable to YES.
```
It seems that sometimes the wrong file type is inferred. I was able to open the image with no problem with gdalinfo, QGIS and custom code. Weirdly the corresponding uncertainty image opened just fine with `ArchGDAL.read`

Limiting the types of drivers that can be used to all calls of `ArchGDAL.read()` fixed the error:

`ArchGDAL.read(FILE_NAME,alloweddrivers =["ENVI"])`

**Comment by pgbrodrick** [comment]
Interesting, thanks for reporting.  I assume that repeat runs on the same image continues to give this error?  If so, any chance you could send me a copy of that input dataset for further testing?

**Comment by adamchlus** [comment]
Yeah, repeat runs resulted in the same error, as did trying to load the image with ArchGDAL in the Julia console, it only worked after specifying drivers. I did do some more testing though on two other machines and only got the error on 1 of the machines, so in total 2 of 3 tested machines didn't work, so maybe its related to underlying dependencies? In any case I'll share the offending image if you want test yourself.
